test-helpers/src/docker_compose.rs

```diff
@@ -81,38 +86,43 @@ impl DockerCompose {
         env::set_var("AWS_SECRET_ACCESS_KEY", "dummy-access-key-secret");
 
         DockerCompose::new("tests/transforms/docker-compose-moto.yml")
-            .wait_for(r#"Press CTRL\+C to quit"#)
-    }
-
-    /// Waits for a string to appear in the docker-compose log output.
-    ///
-    /// This is shorthand for `wait_for_n( log_text, 1 )`
-    ///
-    /// # Arguments
-    /// * `log_text` - A regular expression defining the text to find in the docker-container log
-    /// output.
-    ///
-    /// # Panics
-    /// * If `log_text` is not found within 60 seconds.
-    ///
-    pub fn wait_for(self, log_text: &str) -> Self {
-        self.wait_for_n(log_text, 1)
     }
 
-    /// Waits for a string to appear in the docker-compose log output `count` times.
-    ///
-    /// Counts the number of items returned by `regex.find_iter`.
-    ///
-    /// # Arguments
-    /// * `log_text` - A regular expression defining the text to find in the docker-container log
-    /// output.
-    /// * `count` - The number of times the regular expression should be found.
-    ///
-    /// # Panics
-    /// * If `count` occurrences of `log_text` is not found in the log within 30 seconds.
-    ///
-    pub fn wait_for_n(self, log_text: &str, count: usize) -> Self {
-        self.wait_for_n_t(log_text, count, 30)
+    fn wait_for_containers_to_startup(&self) {
```

Comment:
I'm not a huge fan of this approach as it feels like embedded config. I won't block on this, but we'll probably need a config/file based solution at some point further

Reply:
IMO using an external config file wouldnt give us much here.
The expected workflow is:
1. developer calls DockerCompose::new on a docker-compose.yml that hasnt been configured yet.
2. the test panics with `"Unknown compose file {path} Please implement waiting logic for it here."` and a file + line number in the stacktrace.
3. Developer jumps to the file + line number and adds a call to wait_for_log.

Adding a config file would result in an extra layer of complexity + prevent alternative wait methods if wait_for_log ends up being unsuitable for some case.

That said once we've used this for a while we might uncover reasons why a config solution would be better :shrug: